### PR TITLE
Remove license statements in template files.

### PR DESCRIPTION
**Pull request by Hixie**
It doesn't make sense to put this copyright in these files since the point of these files is to create files for other people's projects.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
I don't understand why but `hostonly_devicelab_tests-2-linux` seems to be repeatedly failing on this PR specifically.

**Comment by Hixie** [comment]
I really don't understand the failure here. It's reliable. The jetifier test fails pretty much every time. All that this PR does is remove some comments from some templates, what could possibly be load-bearing about the comments?

**Comment by jonahwilliams** [comment]
I bet the test is looking for the license files somehow, taking a look now

**Comment by jonahwilliams** [comment]
The reason this is failing is the current logic of:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/android/gradle.dart#L181

This test is attempting to update the settings.gradle, but it will only do so automatically if there is an exact match with a known template.

A reasonable upgrade might be to change this to "exact match, ignoring whitespace and comments". Also fyi @blasten who added this for build aar

**Comment by blasten** [comment]
@Hixie sorry about the trouble. We don't have a good way of migrating and managing files on different versions as lower platforms migrate and abandon patterns. 

The simplest fix is to add another variant  that doesn't include the copyright comment in this file: https://github.com/flutter/flutter/blob/54dba4cd59f4b330e66960f0d37cf57802f675a8/packages/flutter_tools/gradle/deprecated_settings.gradle#L32-L47

**Comment by Hixie** [comment]
I'll definitely add that file, and probably also a test that verifies that the template matches one of those (because the failure mode here is pretty terrible!), but why would we need to patch anything for the test to pass? Doesn't it just use the template? Is the template wrong?

**Comment by blasten** [comment]
@Hixie the tool reads the project's `settings.gradle`, which is the one generated from the template, and then it verifies that the project's `settings.gradle` matches one of the known  and tool-owned versions of this file.  This is done by looking at each entry (separated by `;EOF`) in `flutter/packages/flutter_tools/gradle/deprecated_settings.gradle`. 

If the tool finds one, then it creates a new file `settings_aar.gradle`, which is then used during the build.  This is the mechanism the tool uses to help build apps that have non-androidx plugins.  

Plugins are made visible to Gradle in `settings.gradle`, so to support this workflow it was required to support two different `settings.gradle` files.

This is why the error message:
```
[gradle_jetifier_test] [STDOUT] stdout: [   +3 ms] ✏️  Creating `android/settings_aar.gradle`...
[gradle_jetifier_test] [STDOUT] stdout: [   +1 ms] ✏️  Creating `android/settings_aar.gradle`... (completed in 0ms)
[gradle_jetifier_test] [STDOUT] stdout: [        ] [!] Flutter tried to create the file `android/settings_aar.gradle`, but failed.
[gradle_jetifier_test] [STDOUT] stdout: [        ] To manually update `settings.gradle`, follow these steps:
[gradle_jetifier_test] [STDOUT] stdout: 
[gradle_jetifier_test] [STDOUT] stdout:                1. Copy `settings.gradle` as `settings_aar.gradle`
[gradle_jetifier_test] [STDOUT] stdout:                2. Remove the following code from `settings_aar.gradle`:
[gradle_jetifier_test] [STDOUT] stdout: 
[gradle_jetifier_test] [STDOUT] stdout:                    def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
[gradle_jetifier_test] [STDOUT] stdout:                    def properties = new Properties()
[gradle_jetifier_test] [STDOUT] stdout: 
[gradle_jetifier_test] [STDOUT] stdout:                    assert localPropertiesFile.exists()
[gradle_jetifier_test] [STDOUT] stdout:                    localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }
[gradle_jetifier_test] [STDOUT] stdout: 
[gradle_jetifier_test] [STDOUT] stdout:                    def flutterSdkPath = properties.getProperty("flutter.sdk")
[gradle_jetifier_test] [STDOUT] stdout:                    assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
[gradle_jetifier_test] [STDOUT] stdout:                    apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"

```

Maybe there should have been a check in the test that verifies that the project's `settings.gradle` matches one in `deprecated_settings.gradle`.   As well as rename `deprecated_settings.gradle` since the filename no longer reflects what it contains.  To give you a little context, early on the idea was to build plugins this way (always), and migrate all instances of `settings.gradle` to the new one.  However later on, I found out that this isn't possible.    After chatting with the Gradle team at Google, we found a way to isolate plugins from the app without compromising build speed, devX and maintenance.

**Comment by Hixie** [comment]
Oh I totally missed that because there were compilation errors before that. Do those only appear in the verbose output?

**Comment by blasten** [comment]
@Hixie It shouldn't require verbose output since devicelab tests forward stdout/stderr. I do agree that it's pretty hard to spot failures. I often crash the Chrome tab before I reach the error message. :P. So, I end up downloading the log via curl and then grep. lol

**Comment by Hixie** [comment]
Probably worth discussing what we can do to improve that somewhere.

Anyway. Still planning on fixing all this, just been busy with my Kotlin education.

**Comment by Hixie** [comment]
@jonahwilliams @blasten I updated the legacy file, renamed it, and added a test to verify that the current file is in there. PTAL.

**Review by blasten — APPROVED** [review]
Thanks for fixing this! LGTM

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 1]
```diff
@@ -2484,6 +2484,18 @@ plugin1=${plugin1.path}
       );
     });
   });
+
+  test('Current settings.gradle is in our legacy settings.gradle file set', () {
+    // If this test fails, you probably edited templates/app/android.tmpl.
+    // That's fine, but you now need to add a copy of that file to grad/esettings.gradle.legacy_versions, separated
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/gradle_test.dart:2490]:
```suggestion
    // That's fine, but you now need to add a copy of that file to gradle/settings.gradle.legacy_versions, separated
```

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_tools/test/general.shard/android/gradle_test.dart]: good catch

**Comment by Hixie** [comment]
Thanks for the reviews. Will land on green.

**Pull request merged**